api/auth/external_auth.proto

```diff
@@ -125,7 +119,11 @@ message AttributeContext {
   Peer destination = 2;
 
   // Represents a network request, such as an HTTP request.
-  Request request = 4;
+  oneof req {
+    // HTTPRequest represents the most common form of network request.
+    HTTPRequest request = 3;
```

Comment:
I think we should continue to name this http_request. We are almost immediately going to add tcp_request, don't you think? Try to save typing "http_request" vs. "request" seems kind of silly to me.

Reply:
For developers who write code with IDE, I agree the debate between `http_request` and `request` is silly. For users, operators, admins and others, the saving is quite significant. Istio http attributes are well established, https://istio.io/docs/reference/config/mixer/attribute-vocabulary.html. We can not convince Istio community to change it. Since this is new to Envoy, calling it `request` would not break anything.

For most users, tcp is connection, not a request. I think we better just add `tcp` namespace, which is much shorter than `tcp_request` and more accurate. Once Envoy accepts it, Istio can adopt it as well. By the end of day, Envoy is the first component that generates the attributes, other layers (Envoy filters and Istio) consume the attributes. Let's try out best to align our communities.

Reply:
For Envoy, we have never prioritized convenience of hand authored configurations. The idea is that protos are supposed to be written by machines and expressed to humans via higher-level abstractions like Istio. In this case, I think the request attributes are machine generated (i.e. from Envoy or another data plane proxy), so we should just pick something that is clear at the request level, rather than optimize for saving typing of a few chars here or there.

Reply:
> For Envoy, we have never prioritized convenience of hand authored configurations. 

Agreed. The level to which IMO this PR is being overcomplicated is very surprising to me. This is a straightforward transmission of facts from Envoy to a management server. If the fact language is not optimal as a policy writing language (likely) then your management server (mixer) I think should have its own language and translate.